app/jobs/zip_part_job_base.rb

```diff
@@ -11,8 +11,8 @@ class ZipPartJobBase < ApplicationJob
   end
 
   # Does queue locking on ONLY druid, version and part (as first 3 parameters)
-  def self.lock(*args)
-    "lock:#{name}-#{args.slice(0..2)}"
+  def self.queue_lock_key(*args)
```

Comment:
is this doing the locking, or getting a key?   the name is a bit confusing.  I'd sooner see "queue_lock" with a comment about it returning the key, or about it returning the key as a side effect ... or something like that.

Reply:
this only computes the key for the lock, and is used by other methods (via the superclass in this case) both to try to set the lock, and to clear it.  i'd wanted to avoid just `lock` or `queue_lock` as a name, because both felt ambiguous as to whether they were setting the lock, or returning a lock object (though the method does neither of those things).

i think this ends up being clearer in the base class, with the surrounding context of the other callers and stuff... maybe a better comment here would suffice?  also happy to consider other names if the name itself is what seems really off-base.

Reply:
maybe, if it's not actually doing the locking, just change the comment here (and in other similar spots) to something like "key will ensure we lock on a combination of druid, version, and part (first 3 parameters)" or something like that.   Makes it clear it's just a key, and this method isn't performing the lock.  :-)